### PR TITLE
Add edit profile for returning patients

### LINKED ISSUE #624: Make profile info page editable for returning patients

### PR DESCRIPTION
**Pull request by maya**
## Related Issue or Background Info

- Resolves #624

## TODO

- [x] Allow DOB to be editable #710
- [ ] Add error handling to Patient info form #709
- [x] Add E2E tests  #708

## Changes Proposed

- adds back btn functionality between form page steps
- adds focus to error messages
- updated modal to use flexbox so that it works on mobile
- updates patient form to vertical layout
- fixes some content inconsistencies #289

## Additional Information

- does not include the help link/modal. Possibly do in a separate PR?

## Screenshots / Demos

https://user-images.githubusercontent.com/5249443/106495592-c0482380-6470-11eb-95bd-e34f7d4b0008.mp4

### CODE REVIEW

File: frontend/src/patientApp/timeOfTest/DOB.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 
 import Button from "../../app/commonComponents/Button";
 import TextInput from "../../app/commonComponents/TextInput";
-import { gql, useLazyQuery } from "@apollo/client";
-import { connect, useDispatch, useSelector } from "react-redux";
 import { setPatient } from "../../app/store";
-import { Redirect } from "react-router";
 
 const PATIENT_LINK_VALIDATION_QUERY = gql`
   query PatientLinkVerify($plid: String!, $birthDate: String!) {
```
Comment (emyl3) [inline frontend/src/patientApp/timeOfTest/DOB.tsx:12]: Can we also get back the following fields?       
```
pregnancy
symptoms
firstTest
priorTestDate
priorTestType
priorTestResult
symptomOnset
noSymptoms
```

File: frontend/src/patientApp/timeOfTest/PatientProfileContainer.tsx [hunk 2]
```diff
@@ -1,154 +1,87 @@
...
-  const gender = GENDER_VALUES.find((val) => val.value === props.patient.gender)
-    ?.label;
-
-  const formattedAddress = () => {
-    const lastAddressLine = `${props.patient.city}${
-      props.patient.state && props.patient.city ? "," : ""
-    }${props.patient.city ? " " : ""}${props.patient.state}${
-      props.patient.state ? " " : ""
-    }${props.patient.zipCode}`;
+  history.listen((loc, action) => {
```
Comment (emyl3) [inline frontend/src/patientApp/timeOfTest/PatientProfileContainer.tsx:17]: couldn't figure out a way to make the back button work... except by doing this... 😅

File: frontend/src/patientApp/timeOfTest/PatientProfileFormContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import { useDispatch, connect, useSelector } from "react-redux";
+import { useHistory } from "react-router-dom";
+import { gql, useLazyQuery } from "@apollo/client";
+
+import { setPatient } from "../../app/store";
+import { getPatientLinkIdFromUrl } from "../../app/utils/url";
+import PatientForm from "../../app/patients/PatientForm";
+import PatientTimeOfTestContainer from "../PatientTimeOfTestContainer";
+
+const PATIENT_LINK_VALIDATION_QUERY = gql`
```
Comment (emyl3) [inline frontend/src/patientApp/timeOfTest/PatientProfileFormContainer.tsx:12]: Is there a different way to get the patient data returned after updating it? 🤔

**Comment by maya** [comment]
Note from @emyl3: 
>Also, we should probably hide the DOB field in the patient form so the patient can’t change it because we run into issues when the user tries to update their DOB via the patient app.

**Comment by ClydeDroid** [comment]
Made necessary changes to make DOB editable

**Comment by ClydeDroid** [comment]
E2E tests added!

**Review by nullflux — APPROVED** [review]

**Pull request merged**